python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -124,6 +114,16 @@ def last_materialization_storage_id(self) -> Optional[int]:
         return self.last_materialization_record.storage_id
 
 
+class AssetPartitionRecord(NamedTuple):
+    partition: str
+    last_materialization_storage_id: Optional[int]
+    last_materialization_run_id: Optional[str]
+    last_planned_materialization_storage_id: Optional[int]
+    last_planned_materialization_run_id: Optional[str]
+    last_materialization_failure_storage_id: Optional[int]
```

Comment:
will the event be called "materialization_failure", not "materialization_failed"?

Reply:
that is what I have in https://github.com/dagster-io/dagster/pull/23327 but not landed yet!

Reply:
I based it on RUN_FAILURE, but not held to that

Reply:
"failure" makes sense to me by that logic